castle/cms/browser/content/slideshow.py

```diff
@@ -90,8 +92,19 @@ def get_resource_view_url(self, resource):
             url += '/view'
         return url
 
+    def get_domain(self):
+        portal_url = getToolByName(self.context, "portal_url")
+        portal = portal_url.getPortalObject()
+        full_url = portal.absolute_url()
```

Comment:
just a note could use api.portal.get().absolute_url()

Reply:
that sounds better